source/mainnet/smart-contracts/guides/build-schema.rst

```diff
@@ -40,19 +44,53 @@ For most other cases, it can also be derived automatically, using
    }
 
 Implementing the ``SchemaType`` trait manually only requires specifying one
-function, which is a getter for a ``schema::Type``, which essentially describes
+function (``get_type()``), which is a getter for a ``schema::Type``, which essentially describes
 how this type is represented as bytes and how to represent it.
 
-.. todo::
+For example, the `Cis2 library <https://github.com/Concordium/concordium-rust-smart-contracts/blob/main/concordium-cis2/src/lib.rs>`_
+implements the ``SchemaType`` trait manually for a few types that can not derive their ``SchemaType`` from base types.
+The example below explores the manual ``SchemaType`` trait implementation of the
+``Cis2Event<T, A>``:
+
+.. _build-event-schema:
+
+.. code-block:: rust
+
+   impl<T: IsTokenId, A: IsTokenAmount> schema::SchemaType for Cis2Event<T, A> {
+      fn get_type() -> schema::Type {
+         let mut event_map = BTreeMap::new();
+         event_map.insert(
+            TRANSFER_EVENT_TAG,
+            (
+               "Transfer".to_string(),
+               schema::Fields::Named(vec![
+                  (String::from("token_id"), T::get_type()),
+                  (String::from("amount"), A::get_type()),
+                  (String::from("from"), Address::get_type()),
+                  (String::from("to"), Address::get_type()),
+               ]),
+            ),
+         );
+
+         ...
+
+         schema::Type::TaggedEnum(event_map)
+      }
+   }
+
+.. note ::
 
-   Create an example showing how to manually implement ``SchemaType`` and link
-   to it from here.
+   Event schemas can be represented by the ``TaggedEnum Type`` (`see list of types available to implement a schema <https://docs.rs/concordium-contracts-common/latest/concordium_contracts_common/schema/enum.Type.html>`_).
+   The first byte of the event data logged includes
+   a tag to distinguish the different events on-chain. For example, the TRANSFER_EVENT_TAG is 255 (u8::MAX)
+   which is the key used in the above ``BTreeMap`` to store the ``Transfer`` event schema. The value of the ``Transfer`` event
+   in the above ``BTreeMap`` is a tuple consisting of a ``String`` (named ``Transfer``) and named Fields (``token_id``, ``amount``, ``from``, and ``to``).
```

Comment:
You could mention here that the String is used as the tag in the JSON representation of the value according to the schema.